### PR TITLE
fixes for multiple stratcon issues; objective display improvement

### PR DESCRIPTION
**Pull request by NickAragua**
- Pursuit scenario had the opfor set to run for the opposite edge even though it's the player's goal to get to the opposite edge
- Objective scenarios can sometimes move, but the objective wouldn't go with them
- Now possible to completely fail a facility capture/control objective if the facility is destroyed
- Prevent being able to deploy to invisible objective scenarios from the TO&E
- To help with color-blind users:
o indicates an in-progress objective (not complete, not failed, colored orange)
x indicates a failed objective (no way to complete, red colored)
checkmark indicates a completed objective (green colored)

### CODE REVIEW

File: MekHQ/src/mekhq/campaign/stratcon/StratconStrategicObjective.java [hunk 1]
```diff
@@ -69,6 +70,29 @@ public void setDesiredObjectiveCount(int desiredObjectiveCount) {
...
+        case HostileFacilityControl:
+            // you can fail this by having the facility destroyed
+            StratconFacility hostileFacility = trackState.getFacility(getObjectiveCoords());
+            return hostileFacility == null;
+        case FacilityDestruction:
+            // you can't really permanently fail this
+            return false;
+        default:
+            // we shouldn't be here, but just in case
+            return false;
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/stratcon/StratconStrategicObjective.java:92]: Indent switch

File: MekHQ/src/mekhq/campaign/stratcon/StratconStrategicObjective.java [hunk 2]
```diff
@@ -69,6 +70,29 @@ public void setDesiredObjectiveCount(int desiredObjectiveCount) {
...
+    }
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/stratcon/StratconStrategicObjective.java:93]: Also... this shouldn't line up with the method end

File: MekHQ/src/mekhq/campaign/stratcon/StratconTrackState.java [hunk 3]
```diff
@@ -376,6 +376,34 @@ public StratconCoords findClosestAlliedFacilityCoords(StratconCoords coords) {
...
+                !getObjectivesByCoords().containsKey(destination)) {
+            StratconStrategicObjective objective = getObjectivesByCoords().get(source);
+            // gotta get the cache
+            getObjectivesByCoords().remove(source);
+            getObjectivesByCoords().put(destination, objective);
+            objective.setObjectiveCoords(destination);
+            return true;
+        }
+        
+        return false;
```
Comment (Windchild292) [inline MekHQ/src/mekhq/campaign/stratcon/StratconTrackState.java:395]: Could just be an if else statement

File: MekHQ/src/mekhq/gui/StratconTab.java [hunk 4]
```diff
@@ -291,6 +295,8 @@ private String buildShortStrategicObjectiveText(StratconCampaignState campaignSt
     private String buildStrategicObjectiveText(StratconCampaignState campaignState) {
         StringBuilder sb = new StringBuilder();
 
+        boolean contractIsActive = campaignState.getContract().isActiveOn(getCampaignGui().getCampaign().getLocalDate());
```
Comment (Windchild292) [inline MekHQ/src/mekhq/gui/StratconTab.java:298]: Variable is useless given it is used a grand total of once, and isn't required at all times.

**Review by Windchild292 — COMMENTED** [review]
Couple of things first

**Comment by NickAragua** [comment]
Changes made, plus a data fix for an annoying issue where hostile off-board artillery would count towards primary scenario objectives.

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**